src/AbstractPolyhedron_functions.jl

```diff
@@ -211,3 +212,113 @@ function remove_redundant_constraints(constraints::AbstractVector{LinearConstrai
         return EmptySet{N}()
     end
 end
+
+"""
+    linear_map(M::AbstractMatrix{N},
+               P::AbstractPolyhedron{N};
+               check_invertibility::Bool=true,
+               cond_tol::Number=DEFAULT_COND_TOL,
+               use_inv::Bool=!issparse(M)
+               ) where {N<:Real}
+
+Concrete linear map of a polyhedron in constraint representation.
+
+### Input
+
+- `M` -- matrix
+- `P` -- polyhedron in constraint representation
+- `check_invertibility` -- (optional, deault: `true`) check if the linear map is
+                           invertible, in which case this function uses the matrix
+                           inverse; if the invertibility check fails, or if
+                           this flag is set to `false`, use the vertex representation
+                           to compute the linear map (see below for details)
+- `cond_tol` -- (optional) tolerance of matrix condition (used to check whether
+                the matrix is invertible)
+- `use_inv`  -- (optional, default: `false` if `M` is sparse and `true`
+                otherwise) whether to compute the full left division through
+                `inv(M)`, or to use the left division for each vector; see below 
+
+### Output
+
+The result type depends on the input and the algorithm that was used:
+
+- If the vertex-based approach was used and the dimension of `P` is `2`, the output
+  is a `VPolygon`, otherwise a `VPolytope`.
+- If the input set is an `AbstractPolytope` , then the output is an `HPolytope`.
+- If the input set is an `AbstractPolygon` , then the output is an `HPolygon`.
+- The types of `HalfSpace`, `Hyperplane` and `Line` are preserved.
+- Otherwise, the output is an `HPolyhedron`.
+
+### Algorithm
+
+This function implements two algorithms for the linear map:
+
+- If the matrix ``M`` is invertible (which we check with a sufficient condition),
+  then ``y = M x`` implies ``x = \\text{inv}(M) y`` and we transform the
+  constraint system ``A x ≤ b`` to ``A \\text{inv}(M) y ≤ b``.
+- Otherwise, transform the polyhedron to vertex representation and apply the map
+  to each vertex, returning a polyhedron in vertex representation. 
+
+Note that the vertex representation (second approach) is only available if the
+polyhedron is bounded. Hence we check boundedness first.
+
+To switch off the check for invertibility, set the option
+`check_invertibility=false`. If `M` is not invertible and the polyhedron
+is unbounded, this function returns an exception.
+
+The option `use_inv` lets the user control - in case `M` is invertible - if
+the full matrix inverse is computed, or only the left division on the normal
+vectors. Note that this helps as a workaround when `M` is a sparse matrix, since
+the `inv` function is not available for sparse matrices. In this case, either
+use the option `use_inv=false` or convert the type of `M` as in
+`linear_map(Matrix(M), P)`.
+"""
+function linear_map(M::AbstractMatrix{N},
```

Comment:
Can you add a comment on `_linear_map_hrep` and `_linear_map_vrep`? This is helpful when adding new subtypes to the interface.